### PR TITLE
LPD-21540 - Fix filter actions helper.

### PR DESCRIPTION
**Pull request by juanjofgliferay**
## Task
[LPD-21540](https://liferay.atlassian.net/browse/LPD-21540)

## Issue
It is possible to create StandAlone actions for a Data Set. We were only considering the actions to be lower case.

## Solution
Compare the `permissionKey` from actions created in the Data Set Manager with the `permissionKey` returned in the API using both values in lower case.

### CODE REVIEW

**Comment by juanjofgliferay** [comment]
ci:test:relevant

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/actionItems/filterItemActions.ts [hunk 1]
```diff
@@ -22,12 +22,20 @@ const filterItemActions = (
...
+										action.data?.permissionKey?.toLowerCase()
+								);
+
 								return [
 									...actions,
 									{
 										...action,
 										...itemData.actions[
-											action.data.permissionKey.toLowerCase()
+											matchedPermissionKey[0]
```
Comment (bryceosterhaus) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/actionItems/filterItemActions.ts:38]: Could this technically evaluate to `...itemData.actions[undefined]`. Would be more fail safe to only use `...itemData.actions[matchedPermissionKey[0]]` if `matchedPermissionKey[0]` is not undefined

Comment (juanjofgliferay) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/actionItems/filterItemActions.ts:38]: Based on condition of [L20](https://github.com/liferay-frontend/liferay-portal/pull/4101/files#diff-5c1469eeb87183817f43a118758d7fc81e744f63a65fc3dc5b4120681e8f4403R20), it cannot evaluate to `undefined`.

```
if (action.data?.permissionKey) {
	if (
		itemData.actions &&
		Object.keys(itemData.actions).some(
			(itemAction) =>

                               // if the action name does not match with permissionKey will return false

				itemAction.toLowerCase() ===
				action.data?.permissionKey?.toLowerCase()
		)
```

**Review by markocikos — APPROVED** [review]
@juanjofgliferay LGTM! :sparkles:

**Comment by juanjofgliferay** [comment]
ci:forward

**Comment by juanjofgliferay** [comment]
ci:report:2558548531

**Comment by juanjofgliferay** [comment]
Not related failures based on https://github.com/liferay-frontend/liferay-portal/pull/4101#issuecomment-2037800709

**Comment by juanjofgliferay** [comment]
ci:forward:force

**Pull request closed without merging**